api/internal/authorization/access_control.go

```diff
@@ -110,7 +111,7 @@ func NewRestrictedResourceControl(resourceIdentifier string, resourceType portai
 func DecorateStacks(stacks []portainer.Stack, resourceControls []portainer.ResourceControl) []portainer.Stack {
 	for idx, stack := range stacks {
 
-		resourceControl := GetResourceControlByResourceIDAndType(stack.Name, portainer.StackResourceControl, resourceControls)
+		resourceControl := GetResourceControlByResourceIDAndType(fmt.Sprintf("%d_%s", stack.EndpointID, stack.Name), portainer.StackResourceControl, resourceControls)
```

Comment:
maybe can make use the new stackutils method

Reply:
fixed